### PR TITLE
Add more linters to "lint-changed" workflow

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

File: .lintr_new [hunk 1]
```diff
@@ -25,6 +29,7 @@ linters: linters_with_defaults(
     strings_as_factors_linter(),
     unnecessary_lambda_linter(),
     unneeded_concatenation_linter(allow_single_expression = FALSE),
+    unused_import_linter(),
```
Comment (AshesITR) [inline .lintr_new:32]: Why is that useful to us as a package? It only lints `library()` and friends.

**Comment by AshesITR** [comment]
Also: Is there a reason to let `.lintr` and `.lintr_new` diverge? The split had the goal to eventually use all appropriate linters in both workflows, but to avoid flooding out too many lints that had crept into our code.

File: .lintr_new (hunk 1 above)
Comment (IndrajeetPatil) [reply .lintr_new:32]: It's useful for us to make sure that contributors don't include these lints in their contributions, and the CI/CD catches them even if the reviewers miss it for some reason.

**Comment by IndrajeetPatil** [comment]
> Is there a reason to let .lintr and .lintr_new diverge?

No reason in particular. It's just that running the whole suite of linters included in `.lintr_new` for _all_ files (which is what will happens if we make `.lintr `similar to `.lintr_new`) will be a bit more expensive and time-consuming. OTOH, `.lintr_new` runs all its linters only for the few changed files, and so isn't as expensive. 

At some point, I'd like us to switch to using `linters_with_tags(NULL)`, i.e., include all linters. But this is going to require some groundwork to handle the existing lints.

File: .lintr_new (hunk 1 above)
Comment (AshesITR) [reply .lintr_new:32]: I expect `tests/testthat.R` to produce these lints because it loads some packages without actively using it. Can you confirm this?

**Pull request closed without merging**